clientUtils/Util.ts

```diff
@@ -778,6 +778,20 @@ export function groupMap<T, K>(array: T[], accessor: (v: T) => K): Map<K, T[]> {
     return result
 }
 
+export function keyMap<Key, Value>(
```

Comment:
hm, this is just Lodash's [`_.keyBy`](https://lodash.com/docs/4.17.15#keyBy) but returning a `Map`, right? do we need the output to be a Map?

Reply:
Yes it is just keyBy. In this case, the keys are table identifiers in the explorer config. Often tables don't have an identifier (when there's a single table there is no need to create an identifier), and in that case the key is `undefined`. But a JS object only supports string keys, so the key quietly gets converted to `"undefined"`. 

To avoid thinking about the implications of that, I just went for using a `Map`, which supports `undefined` as a key. 

What do you think? I think we might have had a very similar discussion before when I used a Map again for "defensive" purposes.